src/py/flwr/driver/driver.py

```diff
@@ -126,3 +128,80 @@ def pull_task_res(self, req: PullTaskResRequest) -> PullTaskResResponse:
         # Call Driver API
         res: PullTaskResResponse = self.stub.PullTaskRes(request=req)
         return res
+
+
+class Driver:
+    """`Driver` class provides an interface to the Driver API."""
+
+    def __init__(self) -> None:
+        self.grpc_driver: Optional[GrpcDriver] = None
+        self.workload_id: Optional[int] = None
+        self.task_id_pool: Set[str] = set()
+        self.node = Node(node_id=0, anonymous=True)
+
+    def _check_and_init_grpc_driver(self) -> None:
+        # Check if the GrpcDriver is initialized
+        if self.grpc_driver is not None:
+            return
+
+        # Connect and create workload
+        self.grpc_driver = GrpcDriver()
+        self.grpc_driver.connect()
+        res = self.grpc_driver.create_workload(CreateWorkloadRequest())
+        self.workload_id = res.workload_id
+
+    def get_nodes(self) -> List[Node]:
+        """Get node IDs."""
+        self._check_and_init_grpc_driver()
+
+        # Call GrpcDriver method
+        res = self.grpc_driver.get_nodes(  # type: ignore
+            GetNodesRequest(workload_id=self.workload_id)  # type: ignore
+        )
+        return list(res.nodes)
+
+    def push_task_ins(self, task_ins_list: Iterable[TaskIns]) -> List[str]:
```

Comment:
Let's use `List` here to be consistent with the argument name (we can generalize to `Iterable` later):
```suggestion
    def push_task_ins(self, task_ins_list: List[TaskIns]) -> List[str]:
```